src/templates/index.html

```diff
@@ -66,24 +66,42 @@ <h3>Troubleshooting and Feedback</h3>
     </div>
 </body>
 <script>
-    // generate or retrieve the user_id and created_at timestamp
+    // Generate or retrieve the user_id and created_at timestamp
     let userId = localStorage.getItem('user_id');
     const createdAt = new Date().toLocaleString('en-US', { timeZone: 'America/New_York' });
 
     if (!userId) {
-        userId = crypto.randomUUID(); // generate a unique ID
+        userId = crypto.randomUUID(); // Generate a unique ID
     }
 
     localStorage.setItem('user_id', userId);
     localStorage.setItem('created_at', createdAt);
     console.log(`User ID: ${userId}`);
     console.log(`Created At: ${createdAt}`);
 
-    // send the user_id and created_at timestamp to the server
+    // Increment total_participants for the current day
+    fetch('/track-visit', {
```

Comment:
This endpoint should only be hit when there isn't already a userId in local storage. Can you move this logic to inside the if !userId condition in line 73?